components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/processors/IdPInitLogoutRequestProcessor.java

```diff
@@ -70,6 +74,26 @@ public SAMLSSOReqValidationResponseDTO process(String sessionId, QueryParamDTO[]
 
         init(queryParamDTOs);
 
+        DiagnosticLog.DiagnosticLogBuilder finalizeDiagLogBuilder = null;
+        if (LoggerUtils.isDiagnosticLogsEnabled()) {
+            // Initialize finalizeDiagLogBuilder here to avoid initializing it in every if condition.
+            finalizeDiagLogBuilder = new DiagnosticLog.DiagnosticLogBuilder(
```

Comment:
can't we go with just `diagnosticLogBuilder` for the variable name?

Reply:
This is added to publish the final status of the logout